core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -437,6 +456,7 @@ public PlanOptimizers(
                                 .addAll(new CanonicalizeExpressions(metadata, typeAnalyzer).rules())
                                 .add(new OptimizeRowPattern())
                                 .build()),
+                pushIntoTableScanOptimizer,
```

Comment:
the optimizer just below seems like it simplifies a lot of things, so i'd like to be after that

any reason this is added precisely here?